File: src/org/luminedroid/preferences/colorpicker/SystemSettingColorPickerPreference.java
/*
 * Copyright (C) 2020 crDroid Android Project
 * SPDX-License-Identifier: Apache-2.0
 */

package org.luminedroid.preferences.colorpicker;

import android.content.Context;
import android.util.AttributeSet;
import org.luminedroid.preferences.SystemSettingsStore;

public class SystemSettingColorPickerPreference extends ColorPickerPreference {

  public SystemSettingColorPickerPreference(Context context, AttributeSet attrs, int defStyle) {
    super(context, attrs, defStyle);
    setPreferenceDataStore(new SystemSettingsStore(context.getContentResolver()));
  }

  public SystemSettingColorPickerPreference(Context context, AttributeSet attrs) {
    super(context, attrs);
    setPreferenceDataStore(new SystemSettingsStore(context.getContentResolver()));
  }

  public SystemSettingColorPickerPreference(Context context) {
    super(context, null);
    setPreferenceDataStore(new SystemSettingsStore(context.getContentResolver()));
  }
}

File: src/org/luminedroid/preferences/colorpicker/SystemSettingColorPickerPreference.kt
/*
 * Copyright (C) 2025 LumineDroid
 *
 * SPDX-License-Identifier: Apache-2.0
 */

package org.luminedroid.preferences.colorpicker

import android.content.Context
import android.util.AttributeSet
import org.luminedroid.preferences.SystemSettingsStore

class SystemSettingColorPickerPreference @JvmOverloads constructor(
    context: Context,
    attrs: AttributeSet? = null,
    defStyle: Int = 0
) : ColorPickerPreference(context, attrs, defStyle) {

    init {
        setPreferenceDataStore(SystemSettingsStore(context.contentResolver))
    }
}

